source/common/ssl/context_impl.h

```diff
@@ -37,8 +36,8 @@ struct SslStats {
   ALL_SSL_STATS(GENERATE_COUNTER_STRUCT, GENERATE_GAUGE_STRUCT, GENERATE_TIMER_STRUCT)
 };
 
-typedef CSmartPtr<SSL, SSL_free> SslConPtr;
-typedef CSmartPtr<X509, X509_free> X509Ptr;
+typedef bssl::UniquePtr<SSL> SslConPtr;
+typedef bssl::UniquePtr<X509> X509Ptr;
```

Comment:
Well, now the style guide is just confusing, then. Should we change:

    Smart pointers are type aliased:

to:

    Smart pointers can optionally be type aliased:

?

I'm going to drop the alias here, please don't merge yet.